osu.Game.Rulesets.Osu.Tests/TestSceneSliderInput.cs

```diff
@@ -484,6 +484,50 @@ public void TestRewindHandling()
             AddAssert("no miss judgements recorded", () => judgementResults.All(r => r.Type.IsHit()));
         }
 
+        /// <summary>
+        /// Sliders are common to by 1/2 or 1/4 beat length in order to place the circle on the next beat.
+        /// This tests a user pressing the next circle in the window between the last tick and the end of the slider (<see cref="SliderEventGenerator.TAIL_LENIENCY"/>).
+        /// </summary>
+        [Test]
+        public void TestHitNextCircleDuringTailLeniency()
+        {
+            const double bpm = 240;
+            const double beat_length = 60000 / bpm;
+            const double slider_start = time_slider_start;
+            const double slider_end = slider_start + beat_length;
+            const double last_tick_time = slider_end + SliderEventGenerator.TAIL_LENIENCY;
+            const double next_circle_time = slider_end + beat_length / 4;
+
+            performTest(new List<ReplayFrame>
+                {
+                    new OsuReplayFrame { Position = Vector2.Zero, Actions = { OsuAction.LeftButton }, Time = time_slider_start },
+                    // This frame is weird because the "up" frame can be skipped if the current time passes it (and is thus no longer in an important section).
+                    // So the idea is to instead generate another important frame at the intended time without yet hitting the next circle.
+                    new OsuReplayFrame { Position = new Vector2(100, 0), Actions = { OsuAction.RightButton }, Time = last_tick_time + 5 },
```

Comment:
Maybe something @bdach would be tangentially interested in and confirm behaviour with stable. I had intermittent test failures on this branch by clearing actions on this frame, which would indicate that we're skipping this frame but perhaps we shouldn't?

Reply:
I've also checked what this test case does on stable. Notably the behaviour differs than from what this test looks like but that's because of note lock which prevents hitting the circle. On stable this results in the slider being judged as OK then the hit circle being a MISS, while on lazer the slider is a GREAT (probably because of the alteration of the leniency mechanic to allow the whole 36ms of leniency window rather than a singular spot check at -36ms).

Test files used, for reference: [test.zip](https://github.com/user-attachments/files/20920393/test.zip)

Reply:
I forgot we turned off `FrameAccuratePlayback`. I guess that's probably going to be it.

Not sure how you're getting a 100 on stable, I'm getting 300 + miss, which is as expected. Could be a timing/replay issue.